vortex-array/src/array/sparse/compute/slice.rs

```diff
@@ -75,4 +75,23 @@ mod tests {
 
         assert_eq!(primitive_doubly_sliced.maybe_null_slice::<u32>(), &[13531]);
     }
+
+    #[test]
+    fn slice_partially_invalid() {
+        let values = vec![0u64].into_array();
+        let indices = vec![0u8].into_array();
+
+        // 512 means that you can slice past the valid end of the u8 indices.
+        let sparse = SparseArray::try_new(indices, values, 1000, 999u64.into()).unwrap();
+        let sliced = slice(&sparse, 0, 1000).unwrap();
```

Comment:
previously, slice would attempt to cast 1000 to the `indices.dtype()` which was `u8`, and would fail. now we cast the indices up to `usize` at comparison time